File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.java
class Solution {
    
    public String solution(String X, String Y) {
        
        String answer = "";
        int[] cntx = new int[10];
        int[] cnty = new int[10];
        for(char c : X.toCharArray()) cntx[c - '0']++;
        for(char c : Y.toCharArray()) cnty[c - '0']++;
        int[] cnt = new int[10];
        for(int i = 0; i < 10; i++) cnt[i] = Math.min(cntx[i], cnty[i]);
        
        StringBuilder temp = new StringBuilder();
        for(int i = 9; i >= 0; i--){
            if(i == 0 && temp.length() == 0 && cnt[i] > 0) temp.append(0);
            else {
                for(int j = 0; j < cnt[i]; j++) temp.append(i);
            }
        }
        
        if(temp.length() == 0) answer = "-1";
        else answer = temp.toString();
        return answer;
    }
}

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.py
from collections import Counter
def solution(X, Y):
    cnt = {str(x):0 for x in range(10)}
    alp1 = Counter(X)
    alp2 = Counter(Y)
    for i in range(10):
        key = str(i)
        if key in alp1.keys() and key in alp2.keys():
            cnt[key] = min(alp1[key], alp2[key])
    
    ret = '';
    for i in range(9,-1, -1):
        num = cnt[str(i)]
        if num > 0:
            ret += str(i) * num
    while len(ret) > 1 and ret[0] == '0':
        ret = ret[1:]
    
    return ret if ret else '-1'
        
        
        
    
"""
짝꿍 = X와 Y에서 공통으로 나타나는 숫자로 만들 수 있는 가장 큰 수
1. X Y 짝꿍없으면 짝꿍: -1
2. X Y 짝꿍 0으로 구성 짝꿍:0
"""